tests/StringyTest.php

```diff
@@ -2482,4 +2482,24 @@ public function htmlDecodeProvider()
             array('>', '&gt;'),
         );
     }
+
+    /**
+     * @dataProvider stripPunctuationProvider()
+     */
+    public function testStripPunctuation($expected, $str, $encoding = null)
+    {
+        $stringy = S::create($str, $encoding);
+        $result = $stringy->stripPunctuation();
+        $this->assertStringy($result);
+        $this->assertEquals($expected, $result);
+        $this->assertEquals($str, $stringy);
+    }
+
+    public function stripPunctuationProvider()
+    {
+        return array(
+            array('foo bar ', '!{foo}:,+"`-=\/ (&bar.) @#'),
+            array('fòô bàř 123', '!{fòô}: (bàř)? №1;2--__3%', 'UTF-8'),
```

Comment:
General idea was to be able to wipe out all non alphanumeric chars from string, leaving words and numbers only. I can rename this method somehow, and then dig deeper in the field of punctuations to add relevant methods.